### PR TITLE
feat(pageHeader): rename property to withoutBackground following design review

### PR DESCRIPTION
**Pull request by lee-chase**
Contributes to #1310

Deprecates the `hasBackgroundAlways` in favour of `withoutBackground` the default result remains the same.

#### What did you change?
Deprecated the `hasBackgroundAlways`
Added deprecated test for `hasBackgroundAlways`
Added `withoutBackground` prop and test

#### How did you test and verify your work?
Ran tests and storybook.

### CODE REVIEW

File: packages/cloud-cognitive/src/components/PageHeader/PageHeader.js [hunk 1]
```diff
@@ -869,12 +888,12 @@ PageHeader.propTypes = {
    */
   collapseHeader: PropTypes.bool,
   /**
-   * If `hasCollapseHeaderToggle` and `hasBackgroundAlways` are set then assistive text is required
+   * If `hasCollapseHeaderToggle` and `withoutBackground` are set then assistive text is required
    * for both the expend and collapse states of the button component used.
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/PageHeader/PageHeader.js:892]:
```suggestion
   * If `hasCollapseHeaderToggle` is set and `withoutBackground` is unset then assistive text is
   * required for both the expend and collapse states of the button component used.
```

File: packages/cloud-cognitive/src/components/PageHeader/PageHeader.js [hunk 2]
```diff
@@ -887,23 +906,18 @@ PageHeader.propTypes = {
    */
   disableBreadcrumbScroll: PropTypes.bool,
   /**
-   * If `hasCollapseHeaderToggle` and `hasBackgroundAlways` are set then assistive text is required
+   * If `hasCollapseHeaderToggle` and `withoutBackground` are set then assistive text is required
    * for both the expend and collapse states of the button component used.
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/PageHeader/PageHeader.js:910]:
```suggestion
   * If `hasCollapseHeaderToggle` is set and `withoutBackground` is unset then assistive text is
   * required for both the expend and collapse states of the button component used.
```

File: packages/cloud-cognitive/src/components/PageHeader/PageHeader.js [hunk 3]
```diff
@@ -338,8 +347,8 @@ export let PageHeader = React.forwardRef(
     }, [fullWidthGrid, narrowGrid]);
 
     useEffect(() => {
-      // Determines if the hasBackgroundAlways should be one based on the header height or scroll
-      let result = hasBackgroundAlways ? 1 : 0;
+      // Determines if the withoutBackground should be one based on the header height or scroll
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/PageHeader/PageHeader.js:350]:
```suggestion
      // Determines the appropriate header background opacity based on the header config/height/scroll and the withoutBackground setting
```

File: packages/cloud-cognitive/src/components/PageHeader/PageHeader.js [hunk 4]
```diff
@@ -378,11 +387,11 @@ export let PageHeader = React.forwardRef(
     ]);
 
     useEffect(() => {
-      // only has toggle if requested and has hasBackgroundAlways
+      // only has toggle if requested and has withoutBackground
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/PageHeader/PageHeader.js:390]:
```suggestion
      // only has toggle if requested and does not have withoutBackground
```

File: packages/cloud-cognitive/src/components/PageHeader/PageHeader.js [hunk 5]
```diff
@@ -1031,11 +1045,16 @@ PageHeader.propTypes = {
       asText: PropTypes.string.isRequired,
     }),
   ]),
+  /**
+   * Specifies if the PageHeader should have a background always on and defaults to the preferred `true`.
+   * When false some parts of the header gain a background if they stick to the top of the PageHeader on scroll.
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/PageHeader/PageHeader.js:1050]:
```suggestion
   * Specifies if the PageHeader should appear without a background color, and defaults to the preferred `false` (a background color is shown).
   * Note that when `true` some parts of the header still gain a background if and when they stick to the top of the PageHeader on scroll.
```

File: packages/cloud-cognitive/src/components/PageHeader/PageHeader.test.js [hunk 6]
```diff
@@ -590,6 +596,56 @@ describe('PageHeader', () => {
...
+      <PageHeader
+        data-testid={dataTestId}
+        title={testProps.title}
+        withoutBackground={true}
+      />
+    );
+
+    const header = screen.getByTestId(dataTestId);
+
+    // When no withoutBackground is true this should result in the value 0 for opacity
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/PageHeader/PageHeader.test.js:611]:
```suggestion
    // When withoutBackground is true this should result in the value 0 for opacity
```

File: packages/cloud-cognitive/src/components/PageHeader/PageHeader.test.js [hunk 7]
```diff
@@ -303,6 +303,12 @@ describe('PageHeader', () => {
     expect(
       document.querySelectorAll(`.${blockClass}__title-icon`)
     ).toHaveLength(1);
+
+    // When no withoutBackground is false this should result in the value 0 for opacity
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/PageHeader/PageHeader.test.js:307]:
```suggestion
    // When withoutBackground is false this should result in the value 1 for opacity
```

File: packages/cloud-cognitive/src/components/PageHeader/PageHeader.test.js [hunk 8]
```diff
@@ -590,6 +596,56 @@ describe('PageHeader', () => {
...
+      />
+    );
+
+    for (let i = 0; i < warnings.length; i++) {
+      expect(warn).toBeCalledWith(warnings[i]);
+    }
+
+    const header = screen.getByTestId(dataTestId);
+
+    // When no hasBackgroundAlways is false this should result in the value 0 for opacity
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/PageHeader/PageHeader.test.js:640]:
```suggestion
    // When hasBackgroundAlways is false this should result in the value 0 for opacity
```

**Review by dcwarwick — CHANGES_REQUESTED** [review]

**Comment by lee-chase** [comment]
Thanks, @dcwarwick added a `/falsy` to your `unset` suggestion to read `withoutBackground is unset/falsy`

**Review by dcwarwick — APPROVED** [review]
thanks - LGTM

**Pull request merged**